yt_dlp/extractor/vk.py

```diff
@@ -701,3 +704,139 @@ def _real_extract(self, url):
+            format_id = str_or_none(stream['type'])
+            if format_id in ('hls', 'live_hls', 'live_playback_hls') or '.m3u8' in url:
+                formats.extend(self._extract_m3u8_formats(url, video_id, m3u8_id=format_id, fatal=False))
+            elif format_id in ('dash', 'live_dash', 'live_playback_dash'):
+                formats.extend(self._extract_mpd_formats(url, video_id, mpd_id=format_id, fatal=False))
+            else:
+                formats.append({
+                    'url': url,
+                    'ext': 'mp4',
+                    'format_id': format_id,
+                    **parse_resolution(self._RESOLUTIONS.get(format_id)),
+                })
+        return formats
+
+    def _extract_common_meta(self, stream_info):
+        return traverse_obj(stream_info, {
+            'title': ('title', {str}),
+            'release_timestamp': ('startTime', {int_or_none}),
+            'thumbnail': ('previewUrl', {url_or_none}),
+            'view_count': ('count', 'views', {int_or_none}),
+            'like_count': ('count', 'likes', {int_or_none}),
+            'categories': ('category', 'title', {lambda i: [str_or_none(i)]}),
+        })
+
+
+class VKPlayIE(VKPlayBaseIE):
+    _VALID_URL = r'https?://vkplay\.live/(?P<username>[^/]+)/record/(?P<id>[a-f0-9\-]+)'
+    _TESTS = [{
+        'url': 'https://vkplay.live/zitsmann/record/f5e6e3b5-dc52-4d14-965d-0680dd2882da',
+        'info_dict': {
+            'id': 'f5e6e3b5-dc52-4d14-965d-0680dd2882da',
+            'ext': 'mp4',
+            'title': 'Atomic Heart (пробуем!) спасибо подписчику EKZO!',
+            'uploader': 'ZitsmanN',
+            'uploader_id': 13159830,
+            'release_timestamp': 1683461378,
+            'release_date': '20230507',
+            'thumbnail': r're:https://images.vkplay.live/public_video_stream/record/f5e6e3b5-dc52-4d14-965d-0680dd2882da/preview\?change_time=\d+',
+            'duration': 10608,
+            'view_count': int,
+            'like_count': int,
+            'categories': ['Atomic Heart'],
+        },
+        'params': {'skip_download': 'm3u8'},
+    }]
+
+    def _real_extract(self, url):
+        username, video_id = self._match_valid_url(url).groups()
+
+        initial_state = self._extract_initial_state(url, video_id)
+        record_info = traverse_obj(initial_state, ('record', 'currentRecord', 'data', {dict}))
+        if not record_info:
+            record_info = self._download_json(
+                f'https://api.vkplay.live/v1/blog/{username}/public_video_stream/record/{video_id}', video_id)
+
+        return {
+            'id': video_id,
+            'formats': self._extract_formats(record_info, video_id),
+            **self._extract_common_meta(record_info),
+            **traverse_obj(record_info, {
+                'uploader': ('blog', 'owner', 'nick', {str_or_none}),
+                'uploader_id': ('blog', 'owner', 'id', {int_or_none}),
+                'duration': ('duration', {int_or_none}),
```

Comment:
IMO these could also be moved into `_extract_common_meta`, see my other suggestion.

`uploader_id` should be a string, so we want to use `str_or_none` to stringify potential integer values.

I think for `uploader` we could just use `{str}` instead of `{str_or_none}`, if all we wanna do is check the type. Either is fine though